packages/typescript-estree/tests/test-utils/test-utils.ts

```diff
@@ -139,6 +140,7 @@ export function omitDeep(
             value.push(visit(el, node));
           }
           node[prop] = value;
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- I don't know if it's safe to change this.
```

Comment:
Seems odd to me to leave an _"I don't know if ..."_ in code. I tried removing them locally and some `semanticInfo.test.ts` snapshots broke. I got around it by:

* Changing the type of `oNode` and the return types of `function visit` & `function omitDeep` to `UnknownObject | undefined`
* Removing this `if (isObjectLike(child)) {` from the `else`

WDYT?

Reply:
Hmm, well I'm scared of that since the one usage of `omitDeep` is asserted to be not null https://github.com/typescript-eslint/typescript-eslint/blob/d6c67fe0854ffdbe7ee2623d7ea05be7a04cd1c0/packages/typescript-estree/tests/test-utils/test-utils.ts#L75-L82

But I'm realizing now that this is "just" test code so I feel a little less scared of messing around with it

Reply:
An easier fix for now seems like just turning `isObjectLike` into an ordinary `boolean`-returning function, since its type predicate isn't actually needed in the two places it's used.  🤷 Plus, no runtime changes, yay!

Reply:
🤝 perfect!